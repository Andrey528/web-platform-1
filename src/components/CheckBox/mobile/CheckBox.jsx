import React from 'react';
import block from 'bem-cn';
import PropTypes from 'prop-types';

import checkImg from '../img/check.png';
import './CheckBox.scss';

const CheckBox = ({ checked, callBack, name }) => {
  const b = block('check-box');
  return (
    <div className={b()}>
      <div className={b('view')}>{checked && <img src={checkImg} alt="" />}</div>
      <input
        className={b('input')}
        name={name}
        type="checkbox"
        checked={checked}
        onChange={callBack}
      />
    </div>
  );
};

CheckBox.propTypes = {
  callBack: PropTypes.func.isRequired,
  checked: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
};

export default CheckBox;