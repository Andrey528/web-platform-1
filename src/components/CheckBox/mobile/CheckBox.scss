.check-box {
  width: 1.5rem;
  height: 1.5rem;
  background: #fff;
  border: 2px solid #BCBCBC;
  box-sizing: border-box;
  border-radius: 3px;
  display: flex;
  justify-content: center;
  align-items: center;
  user-select: none;
  flex-shrink: 0;
  * {
    filter: contrast(0);
  }

  &__input {
    display: none;
  }
}