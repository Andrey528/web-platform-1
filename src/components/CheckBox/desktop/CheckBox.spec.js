import React from 'react';
import { shallow } from 'enzyme';

import checkImg from '../img/check.png';
import CheckBox from './CheckBox';

describe('Button desktop tests', () => {
  const props = { checked: true, callBack: jest.fn(), name: 'test' };
  
  describe('CheckBox tests', () => {
    const checkBox = shallow(<CheckBox {...props} />);

    it('render properly', () => {
      expect(checkBox).toMatchSnapshot();
    });

    it('render checked', () => {
      expect(checkBox.find('div').find('input').props().checked).toEqual(props.checked);
    });

    it('render name', () => {
      expect(checkBox.find('input').props().name).toEqual(props.name);
    });
  });

  describe('Callback test', () => {
    it('onChange test', () => {
      const checkBox = shallow(<CheckBox {...props} />);

      checkBox.find('div').find('input').simulate('change', false);
      expect(props.callBack).toBeCalledWith(false);
    });
  });

  describe('Is checked img', () => {
    it('is checked', () => {
      const checkBox = shallow(<CheckBox {...props} />);
      expect(checkBox.find('div').find('img').props().src).toEqual(checkImg);
    });
  });
});