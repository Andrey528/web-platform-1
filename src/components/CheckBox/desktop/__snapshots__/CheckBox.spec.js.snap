// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button desktop tests CheckBox tests render properly 1`] = `
<div
  className="check-box"
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <img
      alt=""
      src="test-file-stub"
    />
  </div>
  <input
    checked={true}
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    name="test"
    onChange={[MockFunction]}
    type="checkbox"
  />
</div>
`;
